content/blog/greenlake-platform-infrastructure-monitoring-with-apache-skywalking.md

```diff
@@ -0,0 +1,138 @@
+
+rm -rf node_exporter-1.4.0-rc.0.linux-amd64.tar.gz
+
+cd node_exporter-1.4.0-rc.0.linux-amd64
+
+cp node_exporter /usr/sbin
+
+cat<<EOF >nodeexporter.service
+
+[Unit]
+Description=Service to start the node_exporter on the node
+[Service]
+Type=Simple
+ExecStart=/usr/sbin/node_exporter
+StandardOutput=syslog
+StandardError=syslog
+SyslogIdentifier=%n
+
+[Install]
+WantedBy=multi-user.target
+
+EOF
+
+
+mv nodeexporter.service /etc/systemd/system
+systemctl daemon-reload
+systemctl enable nodexporter.service
+systemctl start nodexporter.service
+```
+
+Check the metrics endpoint shows metrics exported from the infrastructure node.
+
+```shell
+curl <http://localhost:9100/metrics>
+```
+
+![](/img/node_metrics.png "Node Metrics")
+
+### Step 2- Setup OTel (OpenTelemetry) collector
+
+OTel provides a set of standardized vendor-agnostic SDKs, APIs, and tools for ingesting, transforming, and sending data to an Observability back-end. We can install OTel collector in the same host, which exposes metrics, or we can create central VM/Bare metal instance to receive telemetry information from several infrastructure instances.
+
+Install OTel collector.
+
+```shell
+sudo wget <https://github.com/open-telemetry/opentelemetry-collector-releases/releases/download/v0.54.0/otelcol_0.54.0_linux_amd64.deb>
+
+sudo dpkg -i otelcol_0.54.0_linux_amd64.deb
+
+systemctl status otelcol
+
+# By default, the otelcol systemd service will be started with the --config=/etc/otelcol/config.yaml option after installation.
+```
+
+Create a new otel configuration file (say /etc/otelcol/otel-collector-config.yaml) to link the metrics from host nodes to SkyWalking oap. The orange box in below images must be modified to point to correct infrastructure hosts and SkyWalking DNS or IP address.
+
+![](/img/otel_collector_configuration.png "OTeL Configuration")
+
+Change OTel configuration to newly created OTel configuration file.
+
+```shell
+cd /etc/otelcol
+
+sudo nano otelcol.conf
+
+#change the path to new configuration file we created in the above step and save
+
+
+sudo systemctl restart otelcol
+
+sudo journalctl -u otelcol
+```
+
+### Step 3- Validate the infrastructure host metrics are seen in the SkyWalking UI
+
+![](/img/skywalking_vm_monitoring.png "SkyWalking UI with Infrastructure details")
+
+# **Conclusion**
+
+Infrastructure and application level monitoring gives holistic picture of service reliability. We have seen, how easily we could setup compute infrastructure resource monitoring, running in on- or off premises, using OpenTelemetry and Skywalking. CPU, Memory, and Disk usage trend in APM tool gives monitoring team, with early insight into potential issues in system, which need to be addressed before its faced by customers of applications. Additionally, APM tools often provide trace, logs, metrics monitoring along with alerting, which can be used in conjunction with infrastructure monitoring to have highly reliable service for customers.
```

Comment:
Some minor edits required here. Remember my point about not using "the royal we".
In the first sentence, add the word "a" prior to "holistic".

The second sentence needs some modifications:
"In this post, **I have shown you** how easily **you** can **set up** compute infrastructure resource monitoring, running in on- or **off-prem**ises, using OpenTelemetry and Sky**W**alking."

The third sentence "trend" should be "trends" (plural) and the sentence doesn't read as well as it could. Perhaps it should be stated more like:
 "CPU, Memory, and Disk usage trends showin in the APM tool give the monitoring team early insights into potential system issues that should be addressed before they impact customer applications."

At the very end, please consider rephrasing to:
"...which can be used in conjunction with infrastructure monitoring **tools** to **provide highly reliable services** for customers."